core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -482,6 +495,54 @@ object UnitActions {
         )
     }
 
+    private fun getRepairTurns(unit: MapUnit): Int {
+        val tile = unit.currentTile
+        if (!tile.isPillaged()) return 0
+        if (tile.improvementInProgress == Constants.repair) return tile.turnsToImprovement
+        var repairTurns = tile.ruleset.tileImprovements[Constants.repair]!!.getTurnsToBuild(unit.civInfo, unit)
+
+        val pillagedImprovement = if(tile.improvementIsPillaged) tile.getTileImprovement()!! else tile.ruleset.tileImprovements[tile.roadStatus.name]!!
+        val turnsToBuild = pillagedImprovement.getTurnsToBuild(unit.civInfo, unit)
+        // cap repair to number of turns to build original improvement
+        if (turnsToBuild < repairTurns) repairTurns = turnsToBuild
+        return repairTurns
+    }
+
+    private fun addRepairAction(unit: MapUnit, actionList: ArrayList<UnitAction>) {
+        if (unit.currentTile.ruleset.tileImprovements[Constants.repair] == null) return
+        if (!unit.hasUniqueToBuildImprovements) return
+        if (unit.isEmbarked()) return
+        val tile = unit.getTile()
+        if (!tile.isPillaged()) return
+
+        val couldConstruct = unit.currentMovement > 0
```

Comment:
If isCityCenter then return, don't show a grayed-out action as current configuration does

Reply:
Hm, a city center should never be able to be Pillaged imo. Adding logic to getFoundCityAction() to prevent this